web/yo/app/scripts/controllers/gene.js

```diff
@@ -1538,40 +1538,144 @@ angular.module('oncokbApp')
                 case 'tumor':
-                    var mutationIndex = $scope.gene.mutations.indexOf(mutation);
-                    var tumorIndex = mutation.tumors.indexOf(tumor);
-                    delete $scope.geneStatus[mutationIndex][tumorIndex];
-                    $scope.geneStatus[mutationIndex] = migrateGeneStatusPosition($scope.geneStatus[mutationIndex], tumorIndex);
+                    delete $scope.geneStatus[indices[0]][indices[1]];
+                    $scope.geneStatus[indices[0]] = migrateGeneStatusPosition($scope.geneStatus[indices[0]], indices[1]);
                     mutation.tumors.removeValue(tumor);
                     break;
                 case 'treatment':
-                    var mutationIndex = $scope.gene.mutations.indexOf(mutation);
-                    var tumorIndex = mutation.tumors.indexOf(tumor);
-                    var therapyCategoryIndex = tumor.TI.indexOf(ti);
-                    var treatmentIndex = ti.treatments.indexOf(treatment);
-                    delete $scope.geneStatus[mutationIndex][tumorIndex][therapyCategoryIndex][treatmentIndex];
-                    $scope.geneStatus[mutationIndex][tumorIndex][therapyCategoryIndex] = migrateGeneStatusPosition($scope.geneStatus[mutationIndex][tumorIndex][therapyCategoryIndex], treatmentIndex);
+                    delete $scope.geneStatus[indices[0]][indices[1]][indices[2]][indices[3]];
+                    $scope.geneStatus[indices[0]][indices[1]][indices[2]] = migrateGeneStatusPosition($scope.geneStatus[indices[0]][indices[1]][indices[2]], indices[3]);
                     ti.treatments.removeValue(treatment);
                     break;
                 }
                 _.each(uuids, function(uuid) {
                     $rootScope.reviewMeta.delete(uuid);
                 });
             }
+
+            function getIndex(mutation, tumor, ti, treatment) {
+                var result = [-1, -1, -1, -1]; // Always return four elements array, standing for mutationIndex, tumorIndex, therapyCategoryIndex and treatmentIndex
+
+                if (mutation && tumor && ti && treatment) {
+                    result[0] = $scope.gene.mutations.indexOf(mutation);
+                    result[1] = mutation.tumors.indexOf(tumor);
+                    result[2] = tumor.TI.indexOf(ti);
+                    result[3] = ti.treatments.indexOf(treatment);
+                } else if (mutation && tumor && ti) {
+                    result[0] = $scope.gene.mutations.indexOf(mutation);
+                    result[1] = mutation.tumors.indexOf(tumor);
+                    result[2] = tumor.TI.indexOf(ti);
+                } else if (mutation && tumor) {
+                    result[0] = $scope.gene.mutations.indexOf(mutation);
+                    result[1] = mutation.tumors.indexOf(tumor);
+                } else if (mutation) {
+                    result[0] = $scope.gene.mutations.indexOf(mutation);
+                }
+                return result;
```

Comment:
It might not be a good idea to duplicate the index searching code that much. maybe we can do something similar like:
if(mutation){
  result[0] = $scope.gene.mutations.indexOf(mutation);
}
if(mutation && tumor) {
  result[1] = mutation.tumors.indexOf(tumor);
}
if(mutation && tumor && ti) {
   result[2] = ..
}
if(mutation && tumor && ti && treatment) {
  result[3] = ...
}